arangod/Aql/RemoteExecutor.cpp

```diff
@@ -351,17 +349,22 @@ std::pair<ExecutionState, Result> ExecutionBlockImpl<RemoteExecutor>::shutdown(i
 
     return {ExecutionState::DONE, TRI_ERROR_INTERNAL};
   }
-#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
-  TRI_ASSERT(_didSendShutdownRequest == false);
+
+  // Already sent a shutdown request, but haven't got an answer yet.
+  if (_didSendShutdownRequest) {
+    return {ExecutionState::WAITING, TRI_ERROR_NO_ERROR};
```

Comment:
Do we want to add. traceShutdown here as well?

Reply:
From the code I had locally, I just added the tracing of actual remote requests, not the tracing of `::shutdown()` calls - I thought that'd be enough in most of the cases where shutdown is of interest at all, and doesn't clutter the log as much.